neps/nep-0621.md

```diff
@@ -0,0 +1,497 @@
+    - While possible, this leads to fragmentation, increases integration costs, and reduces composability between protocols.
+2. **Direct ERC-4626 clone Without NEAR Adjustments**
+    - Rejected because NEAR’s asynchronous execution model makes a one-to-one ERC-4626 clone inefficient. This proposal integrates share and vault logic in a single contract to avoid unnecessary cross-contract calls. Key Differences from ERC-4626 includes:
+        - **Deposit & Mint**:
+          Unlike ERC-4626, deposits and mints on NEAR must be handled through the
+          `ft_on_transfer` callback (as defined by NEP-141), rather than direct method calls.
+        - **Async Execution**:
+          Since NEAR contracts execute asynchronously, results from `preview*` and `max*`
+          view methods cannot be guaranteed to match the actual values during transaction
+          execution. Cross-contract calls may be processed in later blocks, leading to
+          differences between simulated values and actual outcomes.
+
+## Future possibilities
+
+### NEP-245 Multi Token Support
+
+Future vault implementations could extend this standard to support NEP-245 Multi Token contracts as underlying asset. We have also created a [minimal implementation](https://github.com/Meteor-Wallet/tokenized-vault-mt-nep-implementation).
+
+### Multi-Asset Vault Extensions
+
+Future extensions could allow vaults to accept multiple assets for deposit and withdrawal. This would enable the standardization of LP vaults.
+
+### Asynchronous Vault Operations
+
+Future vault standards could introduce asynchronous deposit and withdrawal patterns through `request_deposit` and `request_withdraw` functions. This would enable integration with cross-chain protocols and real-world asset protocols.
+
+## Consequences
+
+### Positive
+
+-   Enables a unified, predictable vault interface.
+
+-   Simplifies integration for wallets, DEXs, and aggregators.
+
+-   Improves security through consistent design and accounting.
+
+-   Encourages reuse of tooling, libraries, and audits.
+
+### Neutral
+
+-   Standard defines interface, not yield strategy — implementation remains flexible.
+
+-   Protocols may implement only relevant parts of the interface initially.
+
+### Negative
+
+-   Migration overhead for existing vault implementations to become compliant.
+
+### Backwards Compatibility
+
+-   No breaking changes to NEP-141 itself, but existing vault-like contracts that don’t conform will need to add or rename methods to comply.
+
+-   Share tokens must be NEP-141 compliant, meaning non-NEP-141 share implementations require migration.
+
+## Changelog
+
+### 1.0.0 - Initial Version
+
+> Placeholder for the context about when and who approved this NEP version.
+
+#### Benefits
+
+> List of benefits filled by the Subject Matter Experts while reviewing this version:
+
+-   Benefit 1
+-   Benefit 2
+
+#### Concerns
+
+> Template for Subject Matter Experts review for this version:
```

Comment:
We can just note here that no major concerns were raised, besides the need to need to be careful with the `#security implications` sections described above